ambuda/tasks/ocr.py

```diff
@@ -0,0 +1,102 @@
+"""Background tasks for proofing projects."""
+
+import time
+from typing import Optional
+
+from celery import group
+from celery.result import GroupResult
+
+from ambuda import consts
+from ambuda import database as db
+from ambuda.enums import SitePageStatus
+from ambuda.tasks import app
+from ambuda.tasks.utils import TaskStatus, CeleryTaskStatus
+from ambuda.utils import google_ocr
+from ambuda.utils.assets import get_page_image_filepath
+from ambuda.utils.revisions import add_revision
+from ambuda import queries as q
+from config import create_config_only_app
+
+
+def _run_ocr_for_page_inner(
+    app_env: str,
+    project_slug: str,
+    page_slug: str,
+) -> int:
+    """Must run in the application context."""
+
+    flask_app = create_config_only_app(app_env)
+    with flask_app.app_context():
+        bot_user = q.user(consts.BOT_USERNAME)
+        if bot_user is None:
+            raise ValueError(f'User "{bot_user}" is not defined.')
```

Comment:
Nit: This will just say `'User "None" is not defined.'` — it should probably say "the bot user is not defined", or use `consts.BOT_USERNAME` in place of `bot_user`